packages/fuels-code-gen/src/program_bindings/abigen/logs.rs

```diff
@@ -1,20 +1,15 @@
-use std::collections::HashSet;
-
 use proc_macro2::TokenStream;
 use quote::quote;
 
 use crate::program_bindings::{
-    abi_types::{FullLoggedType, FullTypeDeclaration},
-    resolved_type::resolve_type,
-    utils::single_param_type_call,
+    abi_types::FullLoggedType, resolved_type::TypeResolver, utils::single_param_type_call,
 };
 
 pub(crate) fn logs_lookup_instantiation_code(
     contract_id: Option<TokenStream>,
     logged_types: &[FullLoggedType],
-    shared_types: &HashSet<FullTypeDeclaration>,
 ) -> TokenStream {
-    let resolved_logs = resolve_logs(logged_types, shared_types);
+    let resolved_logs = resolve_logs(logged_types);
```

Comment:
I know this isn't the main point of the PR, but I love that you've removed the need to pass types lookup arg all over the codebase. Super clean!